doc/proposals/cli-ux-phase1.md

```diff
@@ -0,0 +1,121 @@
+tools and documentation that help a user create, develop, test, package, run, and publish an
+operator are spread among many different repositories, making for a steep learning curve for 
+newcomers.
+
+## Goal
+
+Describe an initial set of CLI additions to support a simplified workflow for an Operator SDK user to run an Operator SDK-developed project in a cluster using OLM.
+
+
+## Use cases
+
+1. As an operator developer, I want to deploy OLM to a development cluster.
+2. As an operator developer, I want to create the resources necessary for running my operator via OLM.
+3. As an operator developer, I want to run my operator via OLM.
+
+## Proposed CLI commands
+
+### `operator-sdk alpha`
+
+Many other Kubernetes CLI projects (e.g. `kubebuilder`, `kubectl`, and `kubeadm`) have an `alpha` subcommand tree that is used to introduce new functionality to the CLI while giving users a cue that the functionality is subject to change. This gives maintainers flexibility while iterating on new CLI features.
+
+The general idea is that new subcommand features should be introduced under the `alpha` subcommand (e.g. `operator-sdk alpha my-command`) and then moved to their own top-level commands once the implementation has matured (e.g. `operator-sdk my-command`).
+
+### `operator-sdk alpha olm init`
+
+Deploys the Operator Lifecycle Manager (OLM). The initial proof-of-concept will initialize OLM using manifests from OLM's repository, using the following URL format, based on provided flags:
+
+`https://raw.githubusercontent.com/operator-framework/operator-lifecycle-manager/master/deploy/<cluster-type>/manifests/<olm-version>/`
+
+#### Flags for `olm init`
+
+| Flag             | Type   | Description                                                                                     |
+|------------------|--------|-------------------------------------------------------------------------------------------------|
+| `--cluster-type` | string | The type of cluster to deploy OLM to, one of `upstream`, `ocp`, `okd` (default `upstream`)      |
+| `--olm-version`  | string | The version of OLM to deploy (default `latest`)                                                 |
+| `--kubeconfig`   | string | Path for custom Kubernetes client config file (overrides the default locations)                 |
+
+**Open questions:** 
+1. Do we need to support different cluster types? Or should we just always use `upstream` since OLM is already installed by default in OCP and OKD clusters?
+
+### `operator-sdk alpha olm up`
+
+Creates all necessary resources in the cluster to run the operator via OLM, waits for the operator to be deployed by OLM, and tails the operator log, similar to `operator-sdk up local`. When the user terminates the process or if a timeout or error occurs, all of the created resources will be cleaned up.
+
+#### Prerequisites
+
+1. An operator container image built and pushed to a registry accessible to the cluster.
```

Comment:
I think we need to seriously consider enterprise developers with this command. This means that pushing to a registry and pull from that is sometimes not accessible, and the user will most likely not have access to pull that image into the cluster. What they usually do have access to is the image stream.

Reply:
Perhaps this line is overly specific. Would if be better if it read:

```
1. The operator container image referenced by the CSV is available to the cluster.
```

Put another way, is there anything extra we need to do to enable enterprise developers to make use of this command, or is it just a matter of documentation?